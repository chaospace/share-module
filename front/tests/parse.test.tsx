/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";
import styled from "styled-components";
import "jest-styled-components";

describe("styled-component스타일 파싱 테스트", () => {
    //template literal방식이 아닌 함수 방식을 이용하면 결국 render후 kebab-case로 변경한다.
    const space = () => ({
        borderRadius: '10px',
        fontSize: "19px"
    });
    const Box = styled.div(space);
    Box.displayName = "Box";
    it("render후 스타일은 저절로 케밥케이스로 변경된다.", () => {
        const result = renderer.create(<Box />).toJSON();
        expect(result).toHaveStyleRule("border-radius", "10px");
        expect(result).toMatchSnapshot();
    });
});