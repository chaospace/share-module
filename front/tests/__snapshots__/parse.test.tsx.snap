// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-component스타일 파싱 테스트 render후 스타일은 저절로 케밥케이스로 변경된다. 1`] = `
.c0 {
  border-radius: 10px;
  font-size: 19px;
}

<div
  className="c0"
/>
`;
